pkg/apis/triggers/v1alpha1/event_listener_validation.go

```diff
@@ -44,16 +44,19 @@ func (s *EventListenerSpec) validate(ctx context.Context, el *EventListener) *ap
 		return apis.ErrMissingField("spec.triggers")
 	}
 	clientset := ctx.Value("clientSet").(dynamic.Interface)
-
 	for i, t := range s.Triggers {
+		// Validate that only one of binding or bindings is set
+		if t.DeprecatedBinding != nil && len(t.Bindings) > 0 {
+			return apis.ErrMultipleOneOf(fmt.Sprintf("spec.triggers[%d].binding", i), fmt.Sprintf("spec.triggers[%d].binding", i))
+		}
 		// Validate optional TriggerBinding
-		for _, b := range t.Bindings {
+		for j, b := range t.Bindings {
```

Comment:
i _think_ this happens before `SetDefaults` is called, so we might want to validate the content of `DepreatedBinding` as well?

Reply:
So, that's what I assumed too but it looks like `setDefaults` happens before: https://github.com/tektoncd/triggers/blob/master/vendor/knative.dev/pkg/webhook/webhook.go#L606:L622